lib/http.dart

```diff
@@ -32,7 +32,7 @@ export 'src/streamed_response.dart';
 /// the same server, you should use a single [Client] for all of those requests.
 ///
 /// For more fine-grained control over the request, use [Request] instead.
-Future<Response> head(url, {Map<String, String> headers}) =>
+FutureOr<Response> head(url, {Map<String, String> headers}) =>
```

Comment:
Exactly. So this method should always return `Future`. `FutureOr` implies that it may synchronously return `Response`.